presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -253,6 +254,8 @@ public PlanOptimizers(
                                         new PushLimitThroughSemiJoin(),
                                         new RemoveTrivialFilters(),
                                         new ImplementFilteredAggregations(),
+                                        new ImplementSingleDistinctAggregation(),
```

Comment:
Order between:
```
new ImplementSingleDistinctAggregation(),
new ImplementMultipleDistinctAggregation(),
```
is important here. However, `IterativeOptimizer` doesn't give you (explicitly) any guarantees.
Consider putting those rules in separate `IterativeOptimizer` executions or make `ImplementMultipleDistinctAggregation` not execute on `single distinct aggregation`